slurmutils/core/editor.py

```diff
@@ -43,8 +43,13 @@ def dump(
         group: str | int | None = None,
     ) -> None:
         """Marshal a configuration model into a file."""
-        Path(file).write_text(self.dumps(obj) + "\n")
-        _set_file_permissions(file, mode=mode, user=user, group=group)
+        # Write to a temp file then atomically replace the target file.
+        dst = Path(file)
```

Comment:
[nit]: Typically I prefer the `target` terminology when performing file operations. E.g.:

```suggestion
        target = Path(file)
        swap = target.with_stem("." + target.stem).with_suffix(target.suffix + ".swp")
```

That's mostly just because I don't like abbreviations that much mostly because shortened/abbreviated names can be ambiguous, and you need to read more code to understand the context behind the name. `dst` could stand for destination or the dreaded daylight savings time (dst). 

I also don't like using `tmp` since I find that name often gets relegated to intermediary representation of another variable, so `swap` would be a better name to me since this is the file we're going to "swap" the "target" out with. However, this is a stylistic preference of mine, not a hard and fast rule.

Reply:
Works for me - I've changed the variables to `target` and `swap`